src/compiler/parser.ts

```diff
@@ -3221,6 +3221,37 @@ module ts {
                     }
                 }
 
+                // If encounter "([", this could be the start of an array binding pattern.
+                // Examples:
+                //      ([ x ]) => { }
+                //      ([ x ])
+                if (second === SyntaxKind.OpenBracketToken) {

```

Comment:
True, I picture arrow functions and destructuring parameters to not be very common in conjunction.